subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -30,18 +31,26 @@ class DeviceWorker(
         Executors.newSingleThreadExecutor().asCoroutineDispatcher()
     )
 
-    private var state: State = device.state()
```

Comment:
The problem was here. To calculate device state we use ADB. If anything happens, the worker will fail.

In the `run` function it will wait with proper retries.